solidity/contracts/isms/CctpIsm.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import {Message} from "../libs/Message.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {TokenMessage} from "../token/libs/TokenRouter.sol";
+import {CctpMessageV2} from "../libs/CctpMessageV2.sol";
+import {IMailbox} from "../interfaces/IMailbox.sol";
+import {AbstractCcipReadIsm} from "./ccip-read/AbstractCcipReadIsm.sol";
+import {ICcipReadIsm} from "../interfaces/isms/ICcipReadIsm.sol";
+import {TypedMemView} from "@memview-sol/contracts/TypedMemView.sol";
+import {IMessageTransmitter} from "../interfaces/cctp/IMessageTransmitter.sol";
+
+contract CctpIsm is AbstractCcipReadIsm {
+    using TypeCasts for address;
+    using Message for bytes;
+    using CctpMessageV2 for bytes29;
+
+    string[] urls;
+    IMailbox mailbox;
+    IMessageTransmitter messageTransmitter;
```

Comment:
these should also be `immutable` to avoid incurring SLOAD costs, especially if they cannot be changed